lightclient/src/lib.rs

```diff
@@ -33,8 +33,13 @@ mod platform;
 #[allow(unused_imports)]
 pub use getrandom as _;
 
-pub use client::LightClientRpc;
-pub use smoldot_light::{AddChainConfig, AddChainConfigJsonRpc, ChainId};
+pub use client::{AddedChain, LightClientRpc, RawLightClientRpc};
+pub use smoldot_light::{
+    platform::PlatformRef, AddChainConfig, AddChainConfigJsonRpc, ChainId, Client, JsonRpcResponses,
+};
+
+#[cfg(feature = "native")]
+pub use smoldot_light::platform::default::DefaultPlatform;
```

Comment:
I'm generally thinking if it's better to expose all of the `smoldot` things behind it's own name; when reading the PR I wasn't sure which of the types actually came from `smoldot` and which were our own.

I think the ideal place to be here is:
- `subxt_lightclient` doesn't need to ever be explicitly depended on; it's just internal logic that Subxt leans on and re-exports as needed.
- It's easy to tell which types are from `smoldot_lightclient` and which types are from our crates (this way, it's super obvious in the example for instance that the RawLightClientBuilder is taking `smoldot` args for instance).

So maybe here we could expose all of the smoldot stuff via a `subxt_lightclient::smoldot::...` and perhaps in subxt we expose our own types in `subxt::client::light_client`, and maybe the smoldot types via either `subxt::ext::smoldot` (or `smoldot_light`.. not sure which makes more sense) or maybe `subxt::client::light_client::smoldot` (or `smoldot_light`) if we prefer to export specific types only and want to namespace them into the light client stuff.

Reply:
Yep that makes sense to have different exports.

I modified the exports as:
- subxt_lightclient::smoldot { .. } where smoldot and smoldot_light objects are exported
- then I exported this in subxt::client::light_client::smoldot